### PR TITLE
feat: function to join the LPR2 registers

### LINKED ISSUE #88: Create function: join_lpr2()

### PR DESCRIPTION
**Pull request by lwjohnst86**
Closes #88. Has tests and documentation generated.

I don't know what I did, but I think I made a branch off of the hba1c branch, so this is a stacked PR.

### CODE REVIEW

**Review by signekb — APPROVED** [review]
Very nice! Looks good to me (except there's no overlap of `recnum` in the simulated data - but that's a different issue). 
Should this logic be added to `algorithm.csv` as well?

**Comment by lwjohnst86** [comment]
Right now there is no logic for the algorithm here 😉 so no need to fill in the algorithm file. Only the `include_` and `exclude_` functions contain the actual logic. This and the other future functions are helper functions.

**Comment by signekb** [comment]
Ah! I thought we wanted logic from all functions (including helper functions) in the `algorithms.csv`

**Review by Aastedet — APPROVED** [review]
Added a commit to expand a bit on the test data.

**Review by signekb — COMMENTED** [review]
Don't we need a `recnum` in one of `actual_lpr_diag` and/or `actual_lpr_adm` that's not in the other to test that its actually a full join and not e.g., a left join?

**Comment by Aastedet** [comment]
> Don't we need a `recnum` in one of `actual_lpr_diag` and/or `actual_lpr_adm` that's not in the other to test that its actually a full join and not e.g., a left join?

In the real data there is always a record of each recnum in both _diag and _adm, so it doesn't matter if it's a full/inner/right or left-join in practice. Now that I think about it, I guess that if there was a mismatch/corruption in the input data, I think we would want to only keep the complete records, so I'll change it to inner_join?

File: tests/testthat/test-joins.R [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  recnum = rep(1:6, each = 2),
+  c_diag = 1:12,
+  c_diagtype = rep(c("A", "B"), 6)
+)
+
+actual_lpr_adm <- tibble::tibble(
+  pnr = rep(1:2, 3),
+  recnum = 2:7,
+  c_spec = 1:6,
+  d_inddto = c("20230101", "20220101", "20210101", "20200101", "20190101", "20180101"),
```
Comment (signekb) [inline tests/testthat/test-joins.R:11]: Is `c_pattype` missing here?

Comment (lwjohnst86) [reply tests/testthat/test-joins.R:11]: Um, `c_pattype` isn't in the variable list.

**Comment by lwjohnst86** [comment]
Added some more tests and refined a few things, but nothing major. Merging in.

**Pull request merged**